lib/kv/manager/src/layout.rs

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Block layout management.
+//!
+//! This module provides traits and implementations for managing how blocks
+//! are arranged in storage, including both contiguous and non-contiguous layouts.
+
+pub mod contiguous;
+
+use crate::dtype::DType;
+
+use derive_builder::Builder;
+use thiserror::Error;
+use tracing::instrument;
+use validator::Validate;
+
+/// Errors that can occur during layout operations
+#[derive(Debug, Error)]
+#[allow(missing_docs)]
+pub enum LayoutError {
+    #[error("Invalid configuration: {0}")]
+    InvalidConfig(String),
+
+    #[error("Validation failed: {0}")]
+    ValidationError(#[from] validator::ValidationErrors),
+
+    #[error("Invalid block index: {0}")]
+    InvalidBlockIndex(usize),
+
+    #[error("Invalid layer index: {0}")]
+    InvalidLayerIndex(usize),
+
+    #[error("Operation failed: {0}")]
+    OperationFailed(String),
+}
+
+/// Storage pattern for layers
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum LayerConfiguration {
```

Comment:
Is layer layout better than configuration?

Configuration I think of as a container for multiple options - not a simple enum - but small nit.

```suggestion
pub enum LayerLayout {
```